### ISSUE TITLE
Upgrade path: 1.3.0 to 1.3.1 failed with harbor service exiting with status code 125

### ISSUE THREAD

**Issue by lgayatri** [vmware/vic-product#1351]
Upgrade at scale failed from 1.3 to 1.3.1.

Builds:
1.3.0 OVA: vic-v1.3.0-3033-f8cc7317.ova
1.3.1 OVA: vic-v1.3.1-rc1-3256-b1b156e5.ova

```
root@vic-st-h2-182 [ /etc/vmware/upgrade ]# ./upgrade.sh
Enter vCenter Server FQDN or IP: wdc-stvc3944-146-dhcp15.eng.vmware.com
Enter vCenter Administrator Username: administrator@vsphere.local
Enter vCenter Administrator Password:
If using an external PSC, enter the FQDN of the PSC instance (leave blank otherwise):
If using an external PSC, enter the PSC Admin Domain (leave blank otherwise):

-------------------------
Starting upgrade 2018-01-30 05:41:24 +0000 UTC
Detected old appliance's version as v1.3.0
Detected valid upgrade path. Upgrade will perform data migration, but previous component logs won't be transferred.
If the old appliance's version is not detected correctly, please enter "n" to abort the upgrade and contact VMware support.
Do you wish to proceed with upgrade? [y/n]
y
Continuing with upgrade

Preparing upgrade environment
Disabling and stopping Admiral and Harbor path startup
Finished preparing upgrade environment

-------------------------
Starting Admiral Upgrade 2018-01-30 05:41:24 +0000 UTC

Performing pre-upgrade checks
Starting Admiral upgrade
Updating Admiral configuration

-------------------------
Starting Harbor Upgrade 2018-01-30 05:41:24 +0000 UTC

Performing pre-upgrade checks
No upgrade operations required for upgrade from Harbor harbor-offline-installer-v1.3.0.tgz
[=] Migrating Harbor configuration
[=] Finished migrating Harbor configuration
Enabling and starting Admiral and Harbor path startup
Upgrade script complete. Exiting.
```

upgrade.log is attached.
[upgradeFailed.txt](https://github.com/vmware/vic-product/files/1676575/upgradeFailed.txt)

journalctl -f
```
root@vic-st-h2-182 [ /var/log/vmware ]# journalctl -f
-- Logs begin at Tue 2018-01-30 05:37:55 UTC. --
Jan 30 06:44:20 vic-st-h2-182.eng.vmware.com docker-compose[23705]: Network harbor_notary-mdb not found.
Jan 30 06:44:20 vic-st-h2-182.eng.vmware.com docker-compose[23705]: Removing network harbor_notary-sig
Jan 30 06:44:20 vic-st-h2-182.eng.vmware.com docker-compose[23705]: Network harbor_notary-sig not found.
Jan 30 06:44:20 vic-st-h2-182.eng.vmware.com docker-compose[23705]: Removing network harbor_harbor-clair
Jan 30 06:44:20 vic-st-h2-182.eng.vmware.com docker-compose[23705]: Network harbor_harbor-clair not found.
Jan 30 06:44:20 vic-st-h2-182.eng.vmware.com docker-compose[23705]: Removing network harbor_harbor-notary
Jan 30 06:44:20 vic-st-h2-182.eng.vmware.com docker-compose[23705]: Network harbor_harbor-notary not found.
Jan 30 06:44:20 vic-st-h2-182.eng.vmware.com docker-compose[23734]: No stopped containers
Jan 30 06:44:20 vic-st-h2-182.eng.vmware.com systemd[1]: harbor.service: Unit entered failed state.
Jan 30 06:44:20 vic-st-h2-182.eng.vmware.com systemd[1]: harbor.service: Failed with result 'exit-code'.
Jan 30 06:44:25 vic-st-h2-182.eng.vmware.com systemd[1]: harbor.service: Service hold-off time over, scheduling restart.
Jan 30 06:44:25 vic-st-h2-182.eng.vmware.com systemd[1]: Stopped Harbor Container Registry.
Jan 30 06:44:25 vic-st-h2-182.eng.vmware.com systemd[1]: Started Harbor Container Registry.
Jan 30 06:44:25 vic-st-h2-182.eng.vmware.com start_harbor.sh[23744]: docker: Error response from daemon: Conflict. The container name "/dch-push" is already in use by container "910e578dcda2004715e9fc35cd17004c6cc8cead6fe5517c73082e81463da851". You have to remove (or rename) that container to be able to reuse that name.
Jan 30 06:44:25 vic-st-h2-182.eng.vmware.com start_harbor.sh[23744]: See 'docker run --help'.
Jan 30 06:44:25 vic-st-h2-182.eng.vmware.com systemd[1]: harbor.service: Main process exited, code=exited, status=125/n/a
Jan 30 06:44:26 vic-st-h2-182.eng.vmware.com docker-compose[23756]: Removing network harbor_harbor
Jan 30 06:44:26 vic-st-h2-182.eng.vmware.com docker-compose[23756]: Network harbor_harbor not found.
Jan 30 06:44:26 vic-st-h2-182.eng.vmware.com docker-compose[23756]: Removing network harbor_notary-mdb
Jan 30 06:44:26 vic-st-h2-182.eng.vmware.com docker-compose[23756]: Network harbor_notary-mdb not found.
Jan 30 06:44:26 vic-st-h2-182.eng.vmware.com docker-compose[23756]: Removing network harbor_notary-sig
Jan 30 06:44:26 vic-st-h2-182.eng.vmware.com docker-compose[23756]: Network harbor_notary-sig not found.
...
```
Harbor logs cannot be found at /storage/log
root@vic-st-h2-182 [ /storage/log ]# ls
admiral  lost+found  vic-machine-server

1.3.1 OVA - https://vic-st-h2-182.eng.vmware.com:8282
administrator@vsphere.local/Admin!23

@reasonerjt @mdubya66 @anchal-agrawal

**Comment by reasonerjt** [comment]
@andrewtchin  Seems the `docker-compose down` was called twice?

**Comment by reasonerjt** [comment]
It's unclear to me why harbor.service entered `failed` state, even `docker-compose rm` is called when there's no container it should return `0`
```
Jan 30 06:44:25 vic-st-h2-182.eng.vmware.com start_harbor.sh[23744]: docker: Error response from daemon: Conflict. The container name "/dch-push" is already in use by container "910e578dcda2004715e9fc35cd17004c6cc8cead6fe5517c73082e81463da851". You have to remove (or rename) that container to be able to reuse that name.
Jan 30 06:44:25 vic-st-h2-182.eng.vmware.com start_harbor.sh[23744]: See 'docker run --help'.
```
 In addition, based on the log in OP I don't see the reason why the container `dch-push` is not removed.

**Comment by lgayatri** [comment]
@reasonerjt , I have live testbed. Please feel free to access.

**Comment by stonezdj** [comment]
I found the 403 error in upgrade.log
<img width="1415" alt="screen shot 2018-01-30 at 7 23 17 pm" src="https://user-images.githubusercontent.com/2696760/35563972-85eb43e0-05f3-11e8-928b-686098ffbd4c.png">

The container dch-push, vic-admiral should be removed
<img width="1366" alt="screen shot 2018-01-30 at 7 21 46 pm" src="https://user-images.githubusercontent.com/2696760/35563948-755670e0-05f3-11e8-845e-9a28d7be7469.png">

Seem something wrong with processes before starting harbor.service

**Comment by morris-jason** [comment]
### Fix:
Looks like the temporary registry wasn't up when we tried to push dch in `start_harbor.sh.`
I'm adding a health check before we try to push dch in order to prevent something similar in the future.

### root cause:
dch-push already exists because start_harbor exited before cleaning up it's docker containers.

```
-- Logs begin at Tue 2018-01-30 05:37:55 UTC, end at Tue 2018-01-30 16:51:50 UTC. --
Jan 30 05:44:42 vic-st-h2-182.eng.vmware.com systemd[1]: Started Harbor Container Registry.
Jan 30 05:44:42 vic-st-h2-182.eng.vmware.com start_harbor.sh[2568]: 910e578dcda2004715e9fc35cd17004c6cc8cead6fe5517c7
3082e81463da851
Jan 30 05:44:45 vic-st-h2-182.eng.vmware.com start_harbor.sh[2568]: The push refers to a repository [127.0.0.1:5000/d
efault-project/dch-photon]
Jan 30 05:44:45 vic-st-h2-182.eng.vmware.com start_harbor.sh[2568]: Get http://127.0.0.1:5000/v2/: EOF
Jan 30 05:44:45 vic-st-h2-182.eng.vmware.com systemd[1]: harbor.service: Main process exited, code=exited, st
atus=1/FAILURE
```

**Comment by andrewtchin** [comment]
Resolved in 1.3.1-rc2 with #1385
